src/models.py

```diff
@@ -117,53 +117,51 @@ class Operation(Base):
     entrypoint_id = Column(UUID(as_uuid=True), ForeignKey("entrypoints.id"))
     hash = Column(String)
     status = Column(String)  # TODO Enum
-    created_at = Column(DateTime(timezone=True), default=datetime.datetime.utcnow())
+    created_at = Column(DateTime(timezone=True),
+                        default=datetime.datetime.utcnow())
 
     contract = relationship("Contract", back_populates="operations")
     entrypoint = relationship("Entrypoint", back_populates="operations")
 
 
 # ------- CONDITIONS ------- #
-
+# Condition class with the requirement to limit the number
+# of calls for new users (sponsees)
 
 class Condition(Base):
     __tablename__ = "conditions"
 
     id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
     type = Column(Enum(ConditionType))
-    sponsee_address = Column(
-        String,
-        CheckConstraint(
-            "(type = 'MAX_CALLS_PER_SPONSEE') = (sponsee_address IS NOT NULL)",
-            name="sponsee_address_not_null_constraint",
-        ),
-        nullable=True,
-    )
     contract_id = Column(
         UUID(as_uuid=True),
-        CheckConstraint(
-            "(type = 'MAX_CALLS_PER_ENTRYPOINT') = (contract_id IS NOT NULL)",
-            name="contract_id_not_null_constraint",
-        ),
         ForeignKey("contracts.id"),
         nullable=True,
     )
     entrypoint_id = Column(
         UUID(as_uuid=True),
-        CheckConstraint(
-            "(type = 'MAX_CALLS_PER_ENTRYPOINT') = (entrypoint_id IS NOT NULL)",
-            name="entrypoint_id_not_null_constraint",
-        ),
         ForeignKey("entrypoints.id"),
         nullable=True,
     )
-    vault_id = Column(UUID(as_uuid=True), ForeignKey("credits.id"), nullable=False)
-    max = Column(Integer, nullable=False)
-    current = Column(Integer, nullable=False)
+    vault_id = Column(UUID(as_uuid=True), ForeignKey(
+        "credits.id"), nullable=False)
+    # New field to store the maximum allowed calls per user
+    max_calls_per_user = Column(Integer, nullable=False)
+    # New field to track current calls per user
+    current_calls_per_user = Column(Integer, nullable=False, default=0)
     created_at = Column(
         DateTime(timezone=True), default=datetime.datetime.utcnow(), nullable=False
     )
 
+    # Add a CheckConstraint to ensure consistency between type and associated fields
+    __table_args__ = (
+        CheckConstraint(
+            "(type = 'MAX_CALLS_PER_ENTRYPOINT' AND entrypoint_id IS NOT NULL) \
+          OR (type = 'MAX_CALLS_PER_ENTRYPOINT') AND contract_id IS NOT NULL)",
+            name="type_to_foreign_key_constraint",
+        ),
+    )
+
```

Comment:
Why do you replace two consecutive constraints by a OR? This does not make sense to me.

Also, is there no constraint on max sponsee condition?

Reply:
I reverse it back as before as this [commit](https://github.com/marigold-dev/gas-station/pull/52/commits/2cc8de0bcabc7c6775a160310e3be478ebf7d1e5), 
There is no constraint for max sponsee condition because it does not depend on the sponsee_address, but it is 
at these 2 fields:
```
    # New field to store the maximum allowed calls per user
    max_calls_per_user = Column(Integer, nullable=False)
    # New field to track current calls per user
    current_calls_per_user = Column(Integer, nullable=False, default=0)
```